core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -652,18 +654,21 @@ class UserService(protected val userInfo: UserInfo,
     }
   }
 
-  private def updateBillingAccountInDatabase(billingProjectName: RawlsBillingProjectName, maybeBillingAccountName: Option[RawlsBillingAccountName]): Future[Option[RawlsBillingProject]] = {
+  private def updateBillingAccountInDatabase(billingProjectName: RawlsBillingProjectName, billingAccountName: Option[RawlsBillingAccountName]): Future[Option[RawlsBillingProject]] =
     dataSource.inTransaction { dataAccess =>
-      for {
-        _ <- dataAccess.rawlsBillingProjectQuery.updateBillingAccount(billingProjectName, maybeBillingAccountName, userInfo.userSubjectId)
-        // Since the billing account has been updated, any existing spend configuration is now out of date
-        _ <- dataAccess.rawlsBillingProjectQuery.clearBillingProjectSpendConfiguration(billingProjectName)
-        // if any workspaces failed to be updated last time, clear out the error message so the monitor will pick them up and try to update them again
-        _ <- dataAccess.workspaceQuery.deleteAllWorkspaceBillingAccountErrorMessagesInBillingProject(billingProjectName)
-        maybeBillingProject <- dataAccess.rawlsBillingProjectQuery.load(billingProjectName)
-      } yield maybeBillingProject
+      val F = Applicative[ReadWriteAction]
```

Comment:
I see what this is doing but with the Applicative syntax it feels less readable. Is there is a reason it can't be expressed as a for-comp as before?

Reply:
It could absolutely but I couldn't really find a satisfying way of doing this. The key difference here is that the original for-comp is being executed conditioanlly and wrapping that with another for-comp triggers a few irrational hates of mine!
- checking that an optional is defined in multiple places
- unnecesary re-load of object
- for-comp with two statements (dunno why, I just prefer bind)

```scala
for {
  projectOpt <- load(billingProjectName)
  _ <- when(projectOpt.exists(_.billingAccount != newBillingAccount)) {  // <- check optional.isDefined here
   // original for-comp here
}
} yield projectOpt.map(_.copy(billingAccount = newBillingAccount) // <- check optional.isDefined here
```

Does it feel sufficiently less-readable to you? If so, I'll get over myself and just do it haha!